Automattic/wp-calypso
# Add: Consistent page headings to improve wayfinding

#### Changes proposed in this Pull Request

* This PR adds page headings to screens across all of Calypso, using the existing `<FormattedHeader>` component.
* It also refactors the `<SidebarNavigation>` component (the component on mobile viewports that opens the sidebar navigation) to clarify the functionality.

#### Background
There are many overarching issues with the information architecture (IA) of Calypso. Ultimately, these will need to be fixed to provide a great experience to all of our users. When the overall IA is improved it will help reduce the cognitive load of finding your way around.

Until that happens, however, adding a consistent page heading pattern throughout Calypso can help bridge the gap.

Some issues this approach solves:

* **Context.** Adding consistent page headings provides context to users about where they are in relation to the rest of Calypso. Currently the only real context provided is seeing the section highlighted in the sidebar nav. On mobile, the sidebar is hidden, so this context is non-existent.
* **Accessibility.** When using assistive technology like a screen reader, the page does not have a clear title.

More background about design exploration, proposals, and why this specific development approach was chosen can be found at p9Jlb4-12y-p2

#### Screenshots

Mobile Before | Mobile After
------------ | -------------
<img width="374" alt="Screen Shot 2019-11-07 at 11 44 33 AM" src="https://user-images.githubusercontent.com/448298/68408839-0b7f6380-0154-11ea-974d-2587b1f0d690.png"> | <img width="374" alt="Screen Shot 2019-11-07 at 11 44 07 AM" src="https://user-images.githubusercontent.com/448298/68408832-07534600-0154-11ea-9d5a-f843520fe9e1.png">

Desktop Before | Desktop After
---- | ----
<img width="1092" alt="image" src="https://user-images.githubusercontent.com/448298/68409091-80529d80-0154-11ea-8410-86d5071171f6.png"> | <img width="1109" alt="image" src="https://user-images.githubusercontent.com/448298/68409058-72048180-0154-11ea-8aed-473a57ffab6c.png">


#### Testing instructions

* Checkout this branch.
* Go through each section of My Site(s) and check for a consistent page heading component.
* Make sure all sections have a page heading, including sub-sections and filters.
* Try to break it.
* Check mobile viewports to make sure it all still works and looks ok.

#### Known issues
* This only adds the page heading to each page. Issues like copy duplication when list headers match the page heading will be addressed in future PRs.
* Sections in /me are not touched with this PR
